### ISSUE TITLE
parsing support for hive `insert into` statements where `table` keyword is optional

### ISSUE THREAD

**Issue by krahnikblis** [sqlfluff/sqlfluff#4816]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

parsing fails on Hive insert statements that are missing the `table` keyword. 

### Expected Behaviour

successful parsing of statement. per [Hive documentation](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=82903069#LanguageManualDML-InsertingdataintoHiveTablesfromqueries), "As of Hive [1.1.0](https://issues.apache.org/jira/browse/HIVE-9353) the TABLE keyword is optional." 

### Observed Behaviour

simple Hive insert statement that is valid for Spark was tested, the parser performed as expected when using the Spark dialect, but fails with Hive dialect. i had to go to the Hive documentation to even know that `table` was something it might expect, and when testing the SQL statement with that keyword included, SQLFluff parsed it fine. 

i tried writing a function to add the `table` keyword if it's missing, but i started thinking of all the ways my function would fail (the variety of acceptable Hive keywords like `overwrite` and `directory` and such, newlines and tabs and spacing between keywords and identifiers, what if a comment is between `into` and the schema/table name, and so one) and realized i might break more than i fix... 

### How to reproduce

this fails:
```python
hivesql = """insert into some_schema.target_table
select `col1`,`col2`,`col3`
from some_schema.source_table t1
where col4 is null;"""
linted = sqlfluff.lint(hivesql, dialect="hive") 
linted
```
the linter returns a parsing failure message:
```
[{'line_no': 1,
  'line_pos': 1,
  'code': 'PRS',
  'description': "Line 1, Position 1: Found unparsable section: 'insert into some_schema.target_table\\nsel...'",
  'name': ''}]
```

but this is fine:
```python
hivesql = """insert into table some_schema.target_table
select `col1`,`col2`,`col3`
from some_schema.source_table t1
where col4 is null;"""
linted = sqlfluff.lint(hivesql, dialect="hive") 
linted
```

### Dialect

hive

### Version

sqlfluff.__version__ # '2.0.7'
python version 3.11.3

### Configuration

default config used in example above

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
